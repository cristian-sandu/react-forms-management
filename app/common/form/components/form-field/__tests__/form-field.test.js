import { Input } from 'antd';
import FormProvider from 'common/form/provider/form-provider';
import { shallow } from 'enzyme';
import React from 'react';

import FormField from '../form-field';

describe('<FormField />', () => {
  // Arrange
  const form = {
    getFieldDecorator: jest.fn(),
  };

  it('Renders correctly', () => {
    // Act
    const wrapper = shallow(
      <FormProvider value={form}>
        <FormField id="test_field">
          <Input />
        </FormField>
      </FormProvider>,
    );

    // Assert
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
