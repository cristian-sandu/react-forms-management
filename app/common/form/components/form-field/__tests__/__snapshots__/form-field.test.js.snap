// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField /> Renders correctly 1`] = `
<FormField
  disabled={false}
  id="test_field"
  layout="horizontal"
  required={true}
  requiredMessage=""
  rules={Array []}
>
  <Input
    disabled={false}
    type="text"
  />
</FormField>
`;
